Formula/unisonlang.rb

```diff
@@ -17,27 +17,41 @@ class Unisonlang < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "8e01b2d90970cb66721cf5f5e5135a7867112fe6239250ea2b0cc5eeebb4f33e"
   end
 
-  depends_on "ghc" => :build
+  depends_on "ghc@8.10" => :build
   depends_on "haskell-stack" => :build
-  depends_on "node" => :build
-
-  # stack/ghc currently have a number of issues building for aarch64
-  # https://github.com/unisonweb/unison/issues/3136
-  depends_on arch: :x86_64
+  depends_on "node@16" => :build
 
   uses_from_macos "xz" => :build
   uses_from_macos "zlib"
 
-  resource "codebase-ui" do
-    url "https://github.com/unisonweb/codebase-ui/archive/refs/tags/release/M3.tar.gz"
-    sha256 "84be9135a821615f8fc73a0a894aa46a11c55393c43dc26e16a5ce75f8063012"
-    version "M3"
+  resource "local-ui" do
+    url "https://github.com/unisonweb/unison-local-ui/archive/refs/tags/release/M4.tar.gz"
+    sha256 "5d4b91d071497d8c2924b80fc1025373d7093270e2570a60525df7056646ff23"
+    version "M4"
   end
 
   def install
     jobs = ENV.make_jobs
     ENV.deparallelize
 
+    # Build and install the web interface
+    resource("local-ui").stage do
+      system "npm", "install", *Language::Node.local_npm_install_args
+
+      # Manually run commands from install script
+      # TODO: system "npm", "run", "ui-core:install"
+      system "npx", "--yes", "elm-git-install"
+      uicore = JSON.parse(File.read("elm-stuff/gitdeps/github.com/unisonweb/ui-core/elm.json"))
+      uicore["dependencies"].each do |name, version_range|
+        package = "#{name}@#{version_range.split(".")[0]}"
+        system "npx", "--yes", "elm-json", "install", package, "--yes"
+      end
```

Comment:
Let's see if we can cause same issue by manually running commands and generate some extra output. May need to check if this is doing everything the same as script.

Reply:
Sadly, not much more info.

For Big Sur failure, it got past `elm-git-install` but failed on first `elm-json install` without any output, e.g.
```console
❯ cat 02.npx
2022-09-14 03:27:16 +0000

npx
--yes
elm-git-install

github.com/unisonweb/ui-core => d6d9cd0

❯ cat 03.npx
2022-09-14 03:27:23 +0000

npx
--yes
elm-json
install
elm/browser@1
--yes

```

Other runners have logs like:

```console
❯ cat 02.npx
2022-09-14 03:26:11 +0000

npx
--yes
elm-git-install

github.com/unisonweb/ui-core => d6d9cd0

❯ cat 03.npx
2022-09-14 03:26:15 +0000

npx
--yes
elm-json
install
elm/browser@1
--yes

npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

-- PACKAGE CHANGES READY -------------------------------------------------------

Saved updated elm.json!

❯ cat 04.npx
2022-09-14 03:26:32 +0000

npx
--yes
elm-json
install
elm/core@1
--yes


-- PACKAGE CHANGES READY -------------------------------------------------------

Saved updated elm.json!
```